torch/distributions/categorical.py

```diff
@@ -44,13 +44,15 @@ def __init__(self, probs=None, logits=None):
             self.probs = probs / probs.sum(-1, keepdim=True)
         else:
             self.logits = logits - log_sum_exp(logits)
-        self._num_events = self.probs.size()[-1] if self.probs is not None else self.logits.size()[-1]
-        batch_shape = self.probs.size()[:-1] if probs is not None else self.logits.size()[:-1]
+        self.probs_or_logits = self.probs if probs is not None else self.logits
+        self._num_events = self.probs_or_logits.size()[-1]
+        self.param_shape = self.probs_or_logits.size()
```

Comment:
If you rename `self.probs_or_logits` to `self._param` then this would become `self._param.shape`.

Reply:
Also, this should not be public imo.